src/util/path.js

```diff
@@ -37,36 +36,26 @@ function createLine(data, driveSmoothness) {
 
 /** Recalculates path every time scroller is moved **/
 function changeCenter(progress, slide, data) {
-    let currentSegmentPosition = data[slide].features[0].geometry.coordinates.slice(0, progress);
-
-    let staticLine = (() => {
-        let arr = [];
+    const getCoordinates = d => !d ? [] : d.features[0].geometry.coordinates
 
-        if (slide === 0) {
-            return currentSegmentPosition;
-        }
-
-        for (let i = 0; i <= slide - 1 ; i++) {
-            arr = arr.concat(data[i].features[0].geometry.coordinates);
-        }
-        arr = arr.concat(currentSegmentPosition);
-        return arr;
-    })();
+    const coordinates = [
+        ...data.slice(0, slide).map(getCoordinates).flat(),
+        ...getCoordinates(data[slide]).slice(0, progress)
```

Comment:
Unrelated to your change here I've been thinking that, for the sake of repurposing this for other use cases, it might be nicer if one didn't have to pass a featureCollection with the predefined stops, but rather one line feature that is split up into story stops on the go...what do you think?

Reply:
I think the nicest way to do this would be to actually store path segments in the story json, along with location, pitch and bearing, so that we don't have to guess which part of the path relates to which slide, and also to make pathless slides easy and clean, and also make sure that this code can play any story, not just the current one